### PR TITLE
Provide option to run out of cluster

### PR DESCRIPTION
**Pull request by lucaspalm**
The changes here will give users the ability to install Portieris in an "out-of-cluster" configuration.  It uses the KUBECONFIG env var to specify the kubeconfig for the cluster Portieris needs to manage.  The in-cluster configuration that exists today will still function like normal when there is no KUBECONFIG env var set.

If choosing to run Portieris in an out-of-cluster configuration, these are the required changes/steps you would need to take:
- ensure KUBECONFIG env var is defined in Portieris runtime and it points to a valid kubeconfig file location
- ensure the tls certs are available at `/etc/certs` in new runtime location
- expose Portieris such that it is reachable from the kubernetes apiserver
- change the MutatingWebhookConfiguration to point to your new Portieris URL

Resolves https://github.com/IBM/portieris/issues/180

### CODE REVIEW

File: helpers/kube/kube.go [hunk 1]
```diff
@@ -37,14 +52,24 @@ func GetKubeClient() *kubernetes.Clientset {
...
+	if kubeconfigSet {
+		glog.Info(fmt.Sprintf("KUBECONFIG env variable is set to %s, using this for kube client config", kubeconfig))
+		config, err = clientcmd.BuildConfigFromFlags("", kubeconfig)
+	} else {
+		glog.Info("KUBECONFIG env variable is NOT set, defaulting to in-cluster kube client config")
+		config, err = rest.InClusterConfig()
+	}
 	if err != nil {
-		return nil, err
+		glog.Fatal(err)
```
Comment (jhart1685) [inline helpers/kube/kube.go:68]: I think this line could stay as `return nil, err` 
The [upstream code](https://github.com/IBM/portieris/blob/master/cmd/trust/main.go#L63) checks for the err, and will fatal accordingly.

**Review by jhart1685 — CHANGES_REQUESTED** [review]

**Comment by jhart1685** [comment]
I have requested one change on this. But think it probably needs a review by @molepigeon or @sjhx too.
Presumably the KUBECONFIG environment variable would be passed through to the Portieris deployment via a configmap in some customized yaml?

**Comment by lucaspalm** [comment]
@jhart1685 Correct.  For example, I have it defined in the deployment yaml as an ENV var to the container in my out-of-cluster prototype.  I also volume mounted the kubeconfig file itself to the corresponding file location specified in KUBECONFIG.

File: helpers/kube/kube.go (hunk 1 above)
Comment (lucaspalm) [reply helpers/kube/kube.go:68]: ohhh got it, will fix.  thanks.

**Review by jhart1685 — APPROVED** [review]
This now looks good to me.

**Review by molepigeon — APPROVED** [review]

**Pull request merged**